dashboard/app/models/script.rb

```diff
@@ -1148,50 +1148,63 @@ def fix_lesson_positions
     end
   end
 
-  def clone_migrated_unit(new_name, new_level_suffix: nil, destination_unit_group_name: nil, version_year: nil, family_name:  nil)
+  def clone_migrated_unit(new_name, new_level_suffix: nil, destination_unit_group_name: nil, destination_professional_learning_course: nil, version_year: nil, family_name:  nil)
     raise 'Script name has already been taken' if Script.find_by_name(new_name) || File.exist?(Script.script_json_filepath(new_name))
 
+    if destination_professional_learning_course.nil? && old_professional_learning_course?
+      raise 'Deeper learning courses must be copied to be new deeper learning courses. Include destination_professional_learning_course to set the professional learning course.'
+    end
+
     destination_unit_group = destination_unit_group_name ?
       UnitGroup.find_by_name(destination_unit_group_name) :
       nil
-    raise 'Destination unit group must have a course version' unless destination_unit_group.nil? || destination_unit_group.course_version
+    raise 'Destination unit group must have a course version' unless destination_unit_group.nil? || destination_professional_learning_course.nil? || destination_unit_group.course_version
 
     begin
       ActiveRecord::Base.transaction do
         copied_unit = dup
-        copied_unit.published_state = SharedCourseConstants::PUBLISHED_STATE.in_development
+        copied_unit.published_state = destination_unit_group.nil? ? SharedCourseConstants::PUBLISHED_STATE.in_development : nil
```

Comment:
Could some of these constants be moved down to around line 1190? My understanding is that DLP courses do not have a unit group, but I might be wrong about that

Reply:
Mostly, there are a lot of conditionals here that I'm having trouble understanding and my personal preference would be to group them as much as possible. But that's a personal preference so if it's tricky or you disagree, feel free to leave it :)